File: my_py_pkg/my_py_pkg/robot_news_station.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsStationNode(Node): #Modify Name
    def __init__(self):
        super().__init__("robot_news_station")
        self.declare_parameter("robot_name_", "R2D2")
        self.declare_parameter("timer_", 1.0)

        self.robot_name_ = self.get_parameter("robot_name_").value
        self.timer_ = self.get_parameter("timer_").value

        self.publisher_ = self.create_publisher(String, "topic_name_string", 10)
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot news Station has been started")
        
    def publish_news(self):
        msg = String()
        msg.data = f"Hello,this is {self.robot_name_} from robot news station!"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode() #Modify Name
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
    

File: my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class RobotNewsStationNode : public rclcpp::Node 
{
public:
    RobotNewsStationNode() : Node("robot_news_station")
    {
        this->declare_parameter("robot_name_", "Dalle");
        this->declare_parameter("timer_", 1.0);

        robot_name_ = this->get_parameter("robot_name_").as_string();
        double timer_period = this->get_parameter("timer_").as_double();

        publisher_ = this->create_publisher<example_interfaces::msg::String>("topic_name_string", 10);
        timer_ = this->create_wall_timer(std::chrono::duration<double>(timer_period), std::bind(&RobotNewsStationNode::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Robot News Station has been started");
    }
private:
    void publishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = "Hi, this is "+ robot_name_ + std::string(" from the robot news station.");
        publisher_->publish(msg);
    }

    std::string robot_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}